packages/venia-ui/lib/components/CartPage/PriceAdjustments/GiftOptions.js

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo } from 'react';
+import gql from 'graphql-tag';
+
+import useGiftOptions from '@magento/peregrine/lib/talons/CartPage/GiftOptions/useGiftOptions';
+
+import Checkbox from '../../Checkbox';
+import TextArea from '../../TextArea';
+import { mergeClasses } from '../../../classify';
+
+import defaultClasses from './giftOptions.css';
+
+const GiftOptions = props => {
+    const {
+        includeGiftReceipt,
+        includePrintedCard,
+        giftMessage,
+        toggleIncludeGiftReceiptFlag,
+        toggleIncludePrintedCardFlag,
+        updateGiftMessage
+    } = useGiftOptions({
+        getGiftOptionsQuery: GET_GIFT_OPTIONS_QUERY,
+        saveGiftOptionsQuery: SET_GIFT_OPTIONS_QUERY
+    });
+
+    const classes = useMemo(() => mergeClasses(defaultClasses, props.classes), [
+        props.classes
+    ]);
+
+    return (
+        <div className={classes.root}>
+            <ul>
+                <Checkbox
+                    id="includeGiftReceipt"
+                    field="includeGiftReceipt"
+                    label="Include gift receipt"
+                    fieldState={{
+                        value: includeGiftReceipt
+                    }}
+                    onClick={toggleIncludeGiftReceiptFlag}
+                />
+            </ul>
+            <ul>
+                <Checkbox
+                    id="includePrintedCard"
+                    field="includePrintedCard"
+                    label="Include printed card"
+                    fieldState={{ value: includePrintedCard }}
+                    onClick={toggleIncludePrintedCardFlag}
+                />
+            </ul>
+            <ul>
+                {includePrintedCard && (
+                    <TextArea
+                        id="cardMessage"
+                        field="cardMessage"
+                        placeholder="Enter your message here"
+                        initialValue={giftMessage}
+                        onChange={updateGiftMessage}
+                    />
+                )}
+            </ul>
+        </div>
+    );
+};
+
+/**
+ * Local query. GQL support is not available as of today.
+ *
+ * Once available, we can change the query to match the schema.
+ */
+const GET_GIFT_OPTIONS_QUERY = gql`
+    query getGiftOptions($cart_id: String) {
+        gift_options(cart_id: $cart_id) @client {
```

Comment:
Is it possible to use the `cart` query from the server schema and just use `@client` fields on the custom props? I think that's the approach you'd want to take if you defined your own typedefs.

Reply:
Do you mean to use it as a fragment on the Cart from remote schema? When we switch to GQL support, this will be a fragment instead of a query so I tried to use local fragments but didn't work dude. I had to settle with the query. I am not sure what you mentioned is possible or I am not aware of that. Let me know if I am missing something here.

Reply:
We discussed in person and concluded that getting typeDefs working locally to augment server schema was not a good use of our time :P Apollo was breaking and it doesn't tell you why.